File: WEB/Frontend/src/app/public/models/Forms/LigneInterventionForm.ts
import {PieceForm} from "./PieceForm";

export class LigneInterventionForm {
  idReclamtionOuPrevnetif!:number;
  etat!:number;
  piece!:PieceForm[];


  constructor(idReclamtionOuPrevnetif: number, etat: number, piece?: PieceForm[]) {
    if(piece){
      this.idReclamtionOuPrevnetif = idReclamtionOuPrevnetif;
      this.etat = etat;
      this.piece = piece;
    }
    else
      this.idReclamtionOuPrevnetif = idReclamtionOuPrevnetif;
      this.etat = etat;
  }






}

File: WEB/Backend/src/main/java/com/example/gmaoapp/models/forms/LigneInterventionForm.java
package com.example.gmaoapp.models.forms;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class LigneInterventionForm {
    private Long idReclamtionOuPrevnetif;
    private int etat;
    private List<PieceForm> piece;

}
